src/main/scala/uk/gov/nationalarchives/tdr/api/service/FFIDMetadataService.scala

```diff
@@ -51,6 +63,13 @@ class FFIDMetadataService(ffidMetadataRepository: FFIDMetadataRepository, matche
     }
   }
 
+  private def checkForPasswordProtectedPuids(matchesMetadata: FFIDMetadataInput): String = {
+    matchesMetadata.matches.map(_.puid.getOrElse("")).exists(passwordProtectedPuids.contains) match {
+      case true => PasswordProtected
+      case _ => Success
+    }
+  }
+
```

Comment:
IntelliJ is complaining here because it doesn't like `match` statements with only true and false and prefers if/else. This might be a bit of premature optimisation but would it be worth making it a bit more generic, something like
```scala
private def checkForInvalidPuids(matchesMetadata: FFIDMetadataInput): String = {
    matchesMetadata.matches.map(_.puid.getOrElse("")) match {
      case puids if puids.exists(passwordProtectedPuids.contains) => PasswordProtected
      //we can add more cases here when we're filtering out other files
      case _ => Success
    }
  }
```
Or that might just be confusing and it would make more sense to have them as separate methods. I don't mind so feel free to ignore me.

Reply:
Make sense to me